slurm/slurm.py

```diff
@@ -216,45 +227,44 @@ class DefaultPartitions(SlurmTestBase):
 {{
     "singlenode": "$(getpartitions '-n 4')",
     "multinode": "$(getpartitions '-n 2 -N 2')",
-    "manycores": "$(getpartitions '-n {MAX_CORES_PER_NODE + 1}')",
+    "manycores": "$(getpartitions '-n {max_cores_per_node + 1}')",
     "gpunode": "$(getpartitions '--gpus-per-node=1')"
 }}
 EOF
 """
 
-    @run_after('setup')
-    def get_system(self):
-        self.system = rt.runtime().system.name
-        self.skip_if(self.system == 'manticore', self.descr + ': skipping test on manticore')
-
     @sanity_function
     def assert_partitions(self):
         with open('partitions.json', 'r', encoding='utf-8') as json_file:
             partitions = json.load(json_file)
 
-        return sn.all([
+        asserts = [
             sn.assert_not_found(".", self.stderr, self.descr + ': no error messages'),
             sn.assert_eq(
                 set(PARTITION_MAP[self.system]['smp'] + PARTITION_MAP[self.system]['mpi']),
                 set(partitions['singlenode'].split(',')),
                 self.descr + ': singlenode partitions expected: {0}, found: {1}'
             ),
+            sn.assert_eq(
+                set(x[0] for x in PARTITION_MAP[self.system]['gpu']),
+                set(partitions['gpunode'].split(',')),
+                self.descr + ': gpunode partitions expected: {0}, found: {1}'
+            ),
             sn.assert_eq(
                 set(PARTITION_MAP[self.system]['mpi']),
                 set(partitions['multinode'].split(',')),
                 self.descr + ': multinode partitions expected: {0}, found: {1}'
             ),
+        ]
+        asserts.extend([
             sn.assert_eq(
                 set(PARTITION_MAP[self.system]['mpi']),
                 set(partitions['manycores'].split(',')),
                 self.descr + ': manycores partitions expected: {0}, found: {1}'
             ),
-            sn.assert_eq(
-                set(x[0] for x in PARTITION_MAP[self.system]['gpu']),
-                set(partitions['gpunode'].split(',')),
-                self.descr + ': gpunode partitions expected: {0}, found: {1}'
-            ),
-        ])
+        ]) if self.system != 'manticore' else []
```

Comment:
this is a bit of unintuitive code no? Why not do a simple:
```python
if self.system != 'manticore'
  asserts.extend( ... )
```

Reply:
also, don't these tests now also work for manticore with the changes you made?

Reply:
sure, i just wanted to avoid too much indentation

Reply:
> also, don't these tests now also work for manticore with the changes you made?

no because in `job_submit_manticore.lua` the max number of cores per node is still set to 64.